src/main/java/org/mockito/internal/configuration/injection/SpyOnInjectedFieldsHandler.java

```diff
@@ -28,27 +21,10 @@
 public class SpyOnInjectedFieldsHandler extends MockInjectionStrategy {
 
     @Override
-    protected boolean processInjection(Field field, Object fieldOwner, Set<Object> mockCandidates) {
-        FieldReader fieldReader = new FieldReader(fieldOwner, field);
-
-        // TODO refoctor : code duplicated in SpyAnnotationEngine
-        if(!fieldReader.isNull() && field.isAnnotationPresent(Spy.class)) {
-            try {
-                Object instance = fieldReader.read();
-                if (MockUtil.isMock(instance)) {
-                    // A. instance has been spied earlier
-                    // B. protect against multiple use of MockitoAnnotations.initMocks()
-                    Mockito.reset(instance);
-                } else {
-                    Object mock = Mockito.mock(instance.getClass(), withSettings()
-					    .spiedInstance(instance)
-					    .defaultAnswer(Mockito.CALLS_REAL_METHODS)
-					    .name(field.getName()));
-					setField(fieldOwner, field, mock);
-                }
-            } catch (Exception e) {
-                throw new MockitoException("Problems initiating spied field " + field.getName(), e);
-            }
+    protected boolean processInjection(Field field, Object fieldOwner, Set<Object> ignored) {
+        if (field.isAnnotationPresent(Spy.class)) {
+            Object spy = SpyFieldInitializer.initializeSpy(fieldOwner, field);
+            ignored.add(spy);
```

Comment:
Yes I hesitated, as I wanted to make the second statement a different line. That works for me.